elastalert/alerts.py

```diff
@@ -340,6 +395,71 @@ def __init__(self, rule):
+                    # Log a warning to elastalert saying that we couldn't find that type?
+                    # OR raise and fail to load the alert entirely? Probably the latter...
+                    raise Exception("Could not find a definition for the jira field '{0}'".format(normalized_jira_field))
+                arg_name = field['id']
+                # Check the schema information to decide how to set the value correctly
+                # If it is an array type, it needs to be set in a particular manner:
+                arg_type = None
+                if 'schema' in field and 'type' in field['schema']:
+                    arg_type = field['schema']['type']
+                # Handle arrays of simple types like strings or numbers
+                if arg_type == 'array':
+                    array_items = field['schema']['items']
+                    if array_items == 'string':
+                        # As a convenience, support the scenario wherein the user only provides
+                        # a single value for a multi-value field e.g. jira_labels: Only_One_Label
+                        if type(value) != list:
+                            self.jira_args[arg_name] = [value]
+                        else:
+                            self.jira_args[arg_name] = value
+                    # Also attempt to handle arrays of complex types that have to be passed as objects with an identifier 'key'
+                    elif array_items == 'number':
+                        # As a convenience, support the scenario wherein the user only provides
+                        # a single value for a multi-value field e.g. jira_labels: Only_One_Label
+                        if type(value) != list:
+                            self.jira_args[arg_name] = [int(value)]
+                        else:
+                            self.jira_args[arg_name] = [int(v) for v in value]
+                    else:
+                        # Try setting it as an object, using 'name' as the key
+                        # This may not work, as the key might actually be 'key', 'id', 'value', or something else
+                        # If it works, great!  If not, it will manifest itself as an API error that will bubble up
+                        # Again, as a convenience, support the scenario wherein the user only provides
+                        # a single value for a multi-value field e.g. jira_custom_labels: Only_One_Custom_Label
+                        if type(value) != list:
+                            self.jira_args[arg_name] = [{'name': value}]
+                        else:
+                            self.jira_args[arg_name] = [{'name': v} for v in value]
+                # Handle non-array types
+                else:
+                    # String type
+                    if arg_type == 'string':
+                        self.jira_args[arg_name] = value
+                    # Number type
+                    elif arg_type == 'number':
```

Comment:
FYI, here are all the types that I see defined in our JIRA server:

```
>>> print '\n'.join(sorted(set([b['schema']['type'] for b in j.fields() if 'schema' in b and b['schema']['type'] != 'array'])))
date
datetime
issuetype
number
priority
progress
project
resolution
securitylevel
status
string
timetracking
user
version
```



date and datetime can be passed as simple string values, so long as they parse to dates or datetimes
string can obviously be passed as a simple string value
number can be passed as a simple value, but needs to be an int (and not a string)
securitylevel can be passed with "name": _value_, so long as _value_ is in the supported enum of securitylevel strings in the jira server
user can be passed with "name": _value_, so long as _value_ is a valid user in the jira server

i haven't found a use case for version, timetracking, or progress.  That doesn't mean that the use cases don't exist, I just either wasn't able to surface the type on my test JIRA project, or I couldn't find a meaningful way to set the value.  Either way, I don't think that it should be considered a blocker if we're not able to anticipate and support every single possible type of field for now.  As the need arises, support can be added.